packages/rich-text/src/plugins/List/transforms/toggleList.ts

```diff
@@ -73,30 +76,26 @@ export const toggleList = (editor: PlateEditor, { type }: { type: string }) =>
     } else {
       // selection is a range
 
-      const [startPoint, endPoint] = Range.edges(editor.selection);
-      const commonEntry = Node.common(
-        editor,
-        startPoint.path,
-        endPoint.path
-      ) as NodeEntry<TElement>;
+      const [startPoint, endPoint] = getRangeEdges(editor.selection);
+      const commonEntry = getCommonNode(editor, startPoint.path, endPoint.path) as NodeEntry;
 
       if (
-        listTypes.includes((commonEntry[0] as TElement).type) ||
+        listTypes.includes((commonEntry[0] as Element).type) ||
         commonEntry[0].type === BLOCKS.LIST_ITEM
       ) {
         let listType = commonEntry[0].type;
         if (commonEntry[0].type === BLOCKS.LIST_ITEM) {
-          listType = (Editor.parent(editor, commonEntry[1])[0] as TElement).type;
+          listType = (getParentNode(editor, commonEntry[1] as Location)?.[0] as Element).type;
```

Comment:
```suggestion
          listType = (getParentNode(editor, commonEntry[1] as Location)?.[0] as Element | undefined)?.type;
```

Due the optional chaining the accessed element should be `Element | undefined`, for safety we should also use optional chaining to access the type